scheduler/src/cook/mesos/api.clj

```diff
@@ -526,13 +559,21 @@
                       :commit-latch/uuid (d/squuid)}]
     [commit-latch-id commit-latch]))
 
+(defn- make-partition-ent
+  [partition-type partition]
+  (let [coerce-date (fn coerce-date [date-str]
+                      (tc/to-date (tf/parse partition-date-format date-str)))]
+    (case partition-type
+      "date"
+      {:dataset.partition/begin (coerce-date (partition "begin"))
+       :dataset.partition/end (coerce-date (partition "end"))})))
```

Comment:
What happens if I pass a different partition type?

Reply:
Added a better exception. This should be covered by the validation, but belt and suspenders.